File: Java/백준/Bronze/2309. 일곱 난쟁이/일곱 난쟁이.java
import java.io.*;
import java.util.*;

public class Main {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        int[] dwarfs = new int[9];
        int sum = 0;
        for (int i=0; i<9; i++) {
            dwarfs[i] = Integer.parseInt(br.readLine());
            sum += dwarfs[i];
        }
        
        OUTER_LOOP:
        for (int i=0; i<9; i++) {
            for (int j=i+1; j<9; j++) {
                if (sum - dwarfs[i] - dwarfs[j] == 100) {
                    dwarfs[i]=0;
                    dwarfs[j]=0;
                    break OUTER_LOOP;
                }
            }
        }

        Arrays.sort(dwarfs);
        for(int i=2; i<dwarfs.length; i++) {
            System.out.println(dwarfs[i]);
        }
    }
}

File: C++14/백준/Bronze/2309. 일곱 난쟁이/일곱 난쟁이.cc
#include<bits/stdc++.h>
using namespace std;

int a[9];
int main() {
	for(int i=0; i<9; i++) {
		cin >> a[i];
	}

	for(int i=0; i<8; i++) {
		for(int j=i+1; j<9; j++) {
			int sum=0;
			for (int k=0; k<9; k++) {
				if (k!=i & k!=j) {
					sum += a[k];
				}
			}
			if (sum == 100) {
				a[i]=0;
				a[j]=0;

				sort(a, a+9);
				for (int h : a) {
					if (h!=0){
						cout << h << "\n";
					}
				}
				return 0;
			}
		}
	}
	return 0;
}